File: data/compilers_comparation/src/examples/second_example/src/func41.c
#include "second_example.h" 
array_t* func41(array_t_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array29;
   if (pCounter > 0) {
      array29 = vars->data[--pCounter];
      array29->refC++;
      DEBUG_COPY(array29->id);
   } else {
      array29 = (array_t*)malloc(sizeof(array_t));
      array29->size = 601;
      array29->refC = 1;
      array29->id = 29;
      array29->data = (unsigned int*)malloc(array29->size*sizeof(unsigned int));
      memset(array29->data, 0, array29->size*sizeof(unsigned int));
      DEBUG_NEW(array29->id);
   }
   array_t_param params0;
   params0.size = 1;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   params0.data[0] = array29;
   array_t* array30 = func46(&params0, rng(), loopsFactor);
   DEBUG_RETURN(array30->id);
   free(params0.data);
   unsigned int loop12 = 0;
   unsigned int loopLimit12 = (rand()%loopsFactor)/5 + 1;
   for(; loop12 < loopLimit12; loop12++) {
      for (int i = 0; i < array30->size; i++) {
         array30->data[i]++;
      }
   }
   for (int i = 0; i < array29->size; i++) {
      if (array29->data[i] == 52) { 
         return array29;
      }
   }
   array30->refC--;
   if(array30->refC == 0) {
      free(array30->data);
      free(array30);
      DEBUG_FREE(array30->id);
   }
   return array29;
}

File: data/multilang_comparison/program_nim/src/func41.nim
proc func41*(vars: ptr ArrayTParam, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   var array48: ptr ArrayT
   if pCounter > 0:
      dec pCounter
      array48 = vars.data[pCounter]
      inc array48.refC
      debugCopy(array48.id)
   else:
      array48 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
      array48.size = 353
      array48.refC = 1
      array48.id = 48
      array48.data = cast[ptr UncheckedArray[uint32]](alloc(array48.size * sizeof(uint32)))
      zeroMem(array48.data, array48.size * sizeof(uint32))
      debugNew(array48.id)
   var params0: ArrayTParam
   params0.size = 1
   params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
   params0.data[0] = array48
   var array49: ptr ArrayT = func46(addr params0, getPath(), loopsFactor)
   debugReturn(array49.id)
   dealloc(params0.data)
   let loopLimit17 = (50) div (5) + 1
   for loop17 in 0 ..< loopLimit17:
      discard
      for i in 0 ..< array48.size:
         inc array48.data[i]
   for i in 0 ..< array49.size:
      if array49.data[i] == 19:
         return array49
   dec array49.refC
   if array49.refC == 0:
      dealloc(array49.data)
      dealloc(array49)
      debugFree(array49.id)
   return array48

